### PR TITLE
Sync fork with upstream, keep pinned

### PR DESCRIPTION
**Pull request by nickolasclarke**
This syncs the fork with upstream. They still havent updated requests version, so this pins it again after forcing the sync. sync executed with `gh repo sync --force` and then resolving the merge conflict after to match our pinned version of requests.

oddly, I cannot tag  dwh-reviewers in this repo? idk.

### CODE REVIEW

**Comment by bdewilde** [comment]
What's the context for this, besides "catching back up with the upstream repo"? Was something broken or otherwise bad?

**Comment by nickolasclarke** [comment]
I didnt run into a problem loading for either side as far as I could tell (there are 10ks of records) but looking over the changes since we forked, there were fixes hitting streams we will use. Bumping to the latest seems prudent to me.

File: setup.py [hunk 1]
```diff
@@ -3,30 +3,30 @@
...
+    version="1.0.17",
+    description="Singer.io tap for extracting data from the Zendesk API",
+    author="Twilio",
+    url="https://github.com/twilio-labs/twilio-tap-zendesk",
+    classifiers=["Programming Language :: Python :: 3 :: Only"],
+    py_modules=["tap_zendesk"],
+    install_requires=[
+        "pipelinewise-singer-python==1.2.0",
+        "zenpy==2.0.49",
+        "requests==2.31.0",
```
Comment (bdewilde) [inline setup.py]: While you're here: Do me a favor and bump this to 2.32.0, for [CVE reasons](https://weavegrid.atlassian.net/browse/EX-9839?atlOrigin=eyJpIjoiNWMwMjRiMTBhMGNiNGVhZWI2YTViYzBiNDMwNzcyY2EiLCJwIjoiaiJ9)

**Review by bdewilde — APPROVED** [review]
I'm assuming that these are all upstream changes, and the only custom diff is in `requests`.

File: setup.py (hunk 1 above)
Comment (nickolasclarke) [reply setup.py]: @bdewilde sure thing. How do we do our CVE scans anyhow? I thought about looking that up when I cut this PR.

File: setup.py [hunk 2]
```diff
@@ -3,30 +3,30 @@
...
+    version="1.0.17",
+    description="Singer.io tap for extracting data from the Zendesk API",
+    author="Twilio",
+    url="https://github.com/twilio-labs/twilio-tap-zendesk",
+    classifiers=["Programming Language :: Python :: 3 :: Only"],
+    py_modules=["tap_zendesk"],
+    install_requires=[
+        "pipelinewise-singer-python==1.2.0",
+        "zenpy==2.0.49",
+        "requests==2.32.0",
```
Comment (nickolasclarke) [inline setup.py:16]: only non-upstream change here.

File: setup.py (hunk 1 above)
Comment (bdewilde) [reply setup.py]: I'm sure @susanshrack would be delighted to tell you about the automated CVE scanning system... :)

Comment (susanshrack) [reply setup.py]: Ha 😄 
This confluence page covers it at a high level: https://weavegrid.atlassian.net/wiki/spaces/EN/pages/2038497564/Vuln+Mgmt+Jira+Automation+and+Engineering+Response
TLDR is it's done via Trivy scans triggered via CircleCI jobs - both nightly scans & when merging to the main branch.
